src/main/java/executor/command/CommandEdit.java

```diff
@@ -0,0 +1,312 @@
+            throw new DukeException("Index out of bounds.");
+        }
+    }
+
+    /**
+     * Boolean function to check if input string only contains alphabets.
+     * @param name is the input from user
+     * @return true if input string contains only alphabets
+     */
+    private boolean isAlpha(String name) {
+        return name.matches("[a-zA-Z]+");
+    }
+
+    /**
+     * Function to round up a Double from more than 2 DP to 2 DP.
+     * @param value is the String value given by user
+     * @return is the rounded up value of the user input
+     */
+    private static double round(double value) {
+        BigDecimal bd = BigDecimal.valueOf(value);
+        bd = bd.setScale(2, RoundingMode.HALF_UP);
+        return bd.doubleValue();
+    }
+
+    /**
+     * Function to check if user input has more than 2 decimal places
+     * and if it is more than 2 decimal places it will be rounded up to 2 DP.
+     */
+    private void checkAndChangeToTwoDP() {
+        if (BigDecimal.valueOf(changeValue).scale() > 2) {
+            roundedChangeValue = round(changeValue);
+            moreThanTwoDP = true;
+        }
+    }
+
+    /**
+     * Function to output a String message as a pop-up below GUI when an error is encountered.
+     * @param data is the output message
+     */
+    private void outputError(String data) {
+        this.infoCapsule.setCodeError();
+        this.infoCapsule.setOutputStr(data);
+    }
+
+    /**
+     * Function to output a String message on the GUI when an error is encountered.
+     * @param errorMessage is the output message
+     */
+    private void outputMessageOnGui(String errorMessage) {
+        this.infoCapsule.setCodeCli();
+        this.infoCapsule.setOutputStr(errorMessage);
+    }
+
+    /**
+     * Function to output a String message if the cash value input by user contains more than 2 decimal places.
+     * @return is the output message
+     */
+    private String noteMoreThanTwoDP() {
+        return "\nNOTE : The cash value had more then 2 decimal points, thus it was \nrounded up"
+                + " from "
+                + changeValue
+                + " to "
+                + roundedChangeValue;
+    }
+
+    /**
+     * Function to return a output string if the user input date is in the future.
+     * @return is the output string
+     */
+    private String noteFutureYear() {
+        return "\nNOTE : The year input is in the future";
+    }
+
+    /**
+     * Function to output a String message if the tag input by user contains characters apart from alphabets.
+     * @return is the output message
+     */
+    private String noteHasNonAlphabets() {
+        return "\nNOTE : Tag contains other characters apart from alphabets";
+    }
+
+    /**
+     * Function to output a String message if the tag input by user is empty.
+     * @return is the output message
+     */
+    private String noteTagEmpty() {
+        return "\nNOTE : Tag is empty";
+    }
```

Comment:
abstraction is good and all but couldn't you have just set a variable to equal to the string HAHA